### PR TITLE
Add modified audit suid privilege function rule for CIS

### PR DESCRIPTION
**Pull request by Mackemania**
#### Description:

Add a new rule to accommodate the differences between STIG and CIS regarding execve audit in RHEL8 and RHEL9.

When analyzing a RHEL8 or RHEL9 server with the CIS Workbench after remidiation it fails the settings of `audit execve` because of a difference between STIG and CIS in recommended settings.

#### Review Hints:

The new rule is a modified copy of the rule `audit_rules_suid_privilege_function`.

### CODE REVIEW

**Comment by marcusburghardt** [comment]
Could you provide more details about the differences between STIG and CIS regarding the `audit_rules_suid_privilege_function` rule, please?

**Comment by Mackemania** [comment]
> Could you provide more details about the differences between STIG and CIS regarding the `audit_rules_suid_privilege_function` rule, please?

STIG definition:
```
-a always,exit -F arch=b32 -S execve -C uid!=euid -F euid=0 -k setuid
-a always,exit -F arch=b64 -S execve -C uid!=euid -F euid=0 -k setuid
-a always,exit -F arch=b32 -S execve -C gid!=egid -F egid=0 -k setgid
-a always,exit -F arch=b64 -S execve -C gid!=egid -F egid=0 -k setgid
```

CIS definition
```
-a always,exit -F arch=b32 -C euid!=uid -F auid!=unset -S execve -k user_emulation
-a always,exit -F arch=b64 -C euid!=uid -F auid!=unset -S execve -k user_emulation
```

STIG has focused on both uid and gid while CIS only cares about the uid. The reason why the CIS Workbench fails the analysis is the missing `auid` parameter and the `euid=0`.

**Comment by Mackemania** [comment]
Would it be possible to rerun the checks? `Automatus Fedora / Run Tests (pull_request)` got a http 503 from one of the repos.

**Comment by marcusburghardt** [comment]
> > Could you provide more details about the differences between STIG and CIS regarding the `audit_rules_suid_privilege_function` rule, please?
> 
> STIG definition:
> 
> ```
> -a always,exit -F arch=b32 -S execve -C uid!=euid -F euid=0 -k setuid
> -a always,exit -F arch=b64 -S execve -C uid!=euid -F euid=0 -k setuid
> -a always,exit -F arch=b32 -S execve -C gid!=egid -F egid=0 -k setgid
> -a always,exit -F arch=b64 -S execve -C gid!=egid -F egid=0 -k setgid
> ```
> 
> CIS definition
> 
> ```
> -a always,exit -F arch=b32 -C euid!=uid -F auid!=unset -S execve -k user_emulation
> -a always,exit -F arch=b64 -C euid!=uid -F auid!=unset -S execve -k user_emulation
> ```
> 
> STIG has focused on both uid and gid while CIS only cares about the uid. The reason why the CIS Workbench fails the analysis is the missing `auid` parameter and the `euid=0`.

Thanks for the context. I think this could also be resolved with variables instead of a new rule.
On the other hand, a new rule could also work but I would use a more generic name instead of fixing the benchmark name on it. Supposing the STIG adopts the same approach of CIS in the future, it would be weird to use a `audit_rules_suid_privilege_function_cis` rule for STIG. This is just an example.

Please, consider the introduction of a variable for this rule or renaming the new rule.

**Comment by Mackemania** [comment]
> > > Could you provide more details about the differences between STIG and CIS regarding the `audit_rules_suid_privilege_function` rule, please?
> > 
> > 
> > STIG definition:
> > ```
> > -a always,exit -F arch=b32 -S execve -C uid!=euid -F euid=0 -k setuid
> > -a always,exit -F arch=b64 -S execve -C uid!=euid -F euid=0 -k setuid
> > -a always,exit -F arch=b32 -S execve -C gid!=egid -F egid=0 -k setgid
> > -a always,exit -F arch=b64 -S execve -C gid!=egid -F egid=0 -k setgid
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > CIS definition
> > ```
> > -a always,exit -F arch=b32 -C euid!=uid -F auid!=unset -S execve -k user_emulation
> > -a always,exit -F arch=b64 -C euid!=uid -F auid!=unset -S execve -k user_emulation
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > STIG has focused on both uid and gid while CIS only cares about the uid. The reason why the CIS Workbench fails the analysis is the missing `auid` parameter and the `euid=0`.
> 
> Thanks for the context. I think this could also be resolved with variables instead of a new rule. On the other hand, a new rule could also work but I would use a more generic name instead of fixing the benchmark name on it. Supposing the STIG adopts the same approach of CIS in the future, it would be weird to use a `audit_rules_suid_privilege_function_cis` rule for STIG. This is just an example.
> 
> Please, consider the introduction of a variable for this rule or renaming the new rule.

Thank you! I will change the name, we tried introducing variables, but since the benchmark isn't available when compiling the Jinja macros for the remidiations and documentation it became impossible.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_suid_auid_privilege_function/tests/use_f_key.pass.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+# packages = audit
+
+OTHER_FILTERS=" -C euid!=uid -F auid!=unset"
+
+echo "-a always,exit -F arch=b32 -S execve${OTHER_FILTERS_EUID} -F key=user_emulation" > /etc/audit/rules.d/privileged.rules
+echo "-a always,exit -F arch=b64 -S execve${OTHER_FILTERS_EUID} -F key=user_emulation" >> /etc/audit/rules.d/privileged.rules
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_suid_auid_privilege_function/tests/use_f_key.pass.sh:7]: A wrong variable name cause this test to fail.

```suggestion
echo "-a always,exit -F arch=b32 -S execve${OTHER_FILTERS} -F key=user_emulation" > /etc/audit/rules.d/privileged.rules
echo "-a always,exit -F arch=b64 -S execve${OTHER_FILTERS} -F key=user_emulation" >> /etc/audit/rules.d/privileged.rules
```

**Review by jan-cerny — CHANGES_REQUESTED** [review]

**Comment by jan-cerny** [comment]
The fail on Fedora Rawhide is caused by https://github.com/OpenSCAP/openscap/issues/1995 and isn't related to the contents of this PR.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_suid_auid_privilege_function/tests/use_f_key.pass.sh (hunk 1 above)
Comment (Mackemania) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_suid_auid_privilege_function/tests/use_f_key.pass.sh]: > A wrong variable name cause this test to fail.

Thank you!
The changes have been addressed.

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**